libopae/plugins/xfpga/sysfs.c

```diff
@@ -236,6 +300,14 @@ int sysfs_region_count(void)
 	return _sysfs_region_count;
 }
 
+void sysfs_foreach_region(region_cb cb, void *context)
```

Comment:
Don't you need some locks around this, eg to protect _sysfs_region_count?

Reply:
well, the idea is that the count is only set at initialization, after discovery. I suppose there is a risk of this being called while it is being initialized, in which case I would rather just return from the function instead of blocking. Maybe this function needs a return code?
Another scenario where a lock would help is if we do dynamic updates of region info (like a hotplug event) but we aren't there yet.
Thoughts?

Reply:
I agree that it's probably indicative of a poor design if we were to have thread contention here, but better safe than sorry. :)